Yosemite/Yosemite/Stores/ShippingLabelStore.swift

```diff
@@ -97,6 +122,27 @@ private extension ShippingLabelStore {
         }
     }
 
+    /// Updates/inserts the specified readonly shipping label refund for a shipping label *in a background thread*.
+    /// `onCompletion` will be called on the main thread!
+    func upsertShippingLabelRefundInBackground(siteID: Int64,
+                                               orderID: Int64,
+                                               shippingLabelID: Int64,
+                                               refund: ShippingLabelRefund,
+                                               onCompletion: @escaping () -> Void) {
+        let derivedStorage = sharedDerivedStorage
+        derivedStorage.perform { [weak self] in
+            guard let self = self else { return }
+            guard let shippingLabel = derivedStorage.loadShippingLabel(siteID: siteID, orderID: orderID, shippingLabelID: shippingLabelID) else {
+                return
+            }
```

Comment:
Should we mention why we're not doing anything if the `ShippingLabel` does not exist?

Reply:
Yup, aded a comment in: https://github.com/woocommerce/woocommerce-ios/pull/3294/commits/a272a54b8772fafda0eb7f5c2ea2353db035fbe7